tomviz/DataExchangeFormat.cxx

```diff
@@ -37,23 +37,86 @@ bool DataExchangeFormat::read(const std::string& fileName, vtkImageData* image,
   return GenericHDF5Format::readVolume(reader, deDataNode, image, options);
 }
 
+bool DataExchangeFormat::readDark(const std::string& fileName,
+                                  vtkImageData* image,
+                                  const QVariantMap& options)
+{
+  using h5::H5ReadWrite;
+  H5ReadWrite::OpenMode mode = H5ReadWrite::OpenMode::ReadOnly;
+  H5ReadWrite reader(fileName.c_str(), mode);
+
+  std::string node = "/exchange/data_dark";
+  // If it isn't a data set, we are done
+  if (!reader.isDataSet(node))
+    return false;
+
+  return GenericHDF5Format::readVolume(reader, node, image, options);
+}
+
+bool DataExchangeFormat::readWhite(const std::string& fileName,
+                                   vtkImageData* image,
+                                   const QVariantMap& options)
+{
+  using h5::H5ReadWrite;
+  H5ReadWrite::OpenMode mode = H5ReadWrite::OpenMode::ReadOnly;
+  H5ReadWrite reader(fileName.c_str(), mode);
+
+  std::string node = "/exchange/data_white";
+  // If it isn't a data set, we are done
+  if (!reader.isDataSet(node))
+    return false;
+
+  return GenericHDF5Format::readVolume(reader, node, image, options);
```

Comment:
These two methods are pretty much the same, can we refactor them into a single method that take the path to read the volume from, something like this ...

```c++
bool DataExchangeFormat::readData(const std::string& fileName, 
                              const std::string& path, 
                              vtkImageData* image, 
                              const QVariantMap& options
{
// Read the volume at this path
}

bool DataExchangeFormat::readWhite(const std::string& fileName,
                                   vtkImageData* image,
                                   const QVariantMap& options)
{
  return this->readData(fileName, "/exchange/data_white", image, options);
}

```

Reply:
Done.